enumerables.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |item|
+      result = false unless yield(item)
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each do |item|
+      result = true if yield(item)
+    end
+    result
+  end
+
+  def my_none?
+    result = false
+    my_each do |item|
+      result = true unless yield(item)
+    end
+  end
+
+  def my_count
+    return length unless block_given?
+
+    result = 0
+    my_each do |item|
+      result = true unless yield(item)
+    end
+    result
+  end
+
+  def my_map(proc = nil)
+    result = []
+    my_each do |item|
+      result << if block_given?
+                  yield(item)
+                else
+                  proc.call(item)
+                end
+    end
+    result
+  end
```

Comment:
- [ ] should have working `my_map` **(covered by automated tests)** identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should return an enumerator if no block or argument is given.